.github/workflows/dev.yml

```diff
@@ -1,11 +1,11 @@
 name: dev
 on: 
   push:
-    branches-ignore: 
-      - 'cesium.com'
-      - production
+    branches: 
+      - main
+  pull_request:
```

Comment:
I don't know enough about GHA but I just want to confirm this is actually what we want to do. There's a warning in the [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) saying that:
> workflows that are triggered by the `pull_request_target` event, the `GITHUB_TOKEN` is granted read/write repository permission unless the `permissions` key is specified and the workflow can access secrets, even when it is triggered from a fork.

It also says these are run in the context of the base branch. Does that mean it's trying to access the secrets from the PR repo's fork? or giving access to our secrets from our repo?

Reply:
That is true for the `pull_request_target` event.  This is the `pull_request` event, which only grants those permissions to _internal_ contributors, not external contributors.

For external contributors, the docs say the following:

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository. The GITHUB_TOKEN has read-only permissions in pull requests from forked repositories. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)."